libs/vgc/ui/menubutton.h

```diff
@@ -0,0 +1,246 @@
+    }
+
+    Label* shortcutLabel() const {
+        return shortcutLabel_;
+    }
+
+    void setActive(bool isActive);
+
+private:
+    // Components
+    Action* action_ = nullptr;
+    Widget* iconWidget_ = nullptr;
+    Label* textLabel_ = nullptr;
+    Label* shortcutLabel_ = nullptr;
+
+    // Style
+    bool isActive_ = false;
+
+    // Background
+    graphics::GeometryViewPtr triangles_;
+    bool reload_ = true;
+
+    // Behavior
+    bool tryClick_();
+    void onActionChanged_();
+
+    VGC_SLOT(onActionChangedSlot_, onActionChanged_);
+    VGC_SLOT(onActionAboutToBeDestroyed_, destroy);
+};
+
+/// \enum vgc::ui::MenuDropDirection
+/// \brief The direction in which a dropdown menu should appear.
+///
+enum class MenuDropDirection {
+    Horizontal,
+    Vertical,
+};
+
+/// \class vgc::ui::MenuButton
+/// \brief A button with a special layout for Menus.
+///
+class VGC_UI_API MenuButton : public ActionButton {
+private:
+    VGC_OBJECT(MenuButton, ActionButton)
+    friend Menu;
+
+protected:
+    MenuButton(Action* action, FlexDirection layoutDirection);
+
+public:
+    /// Creates an MenuButton with the given `action`.
+    ///
+    static MenuButtonPtr
+    create(Action* action, FlexDirection layoutDirection = FlexDirection::Column);
+
+    MenuDropDirection menuDropDir() const {
+        return menuDropDir_;
+    }
+
+    geometry::Vec2f preferredIconSize() const {
+        return iconWidget()->preferredSize();
+    }
+
+    geometry::Vec2f preferredTextSize() const {
+        return textLabel()->preferredSize();
+    }
+
+    geometry::Vec2f preferredShortcutSize() const {
+        return shortcutLabel()->preferredSize();
+    }
+
+    geometry::Vec2f preferredArrowSize() const {
+        // XXX todo
+        return isDisabled() ? geometry::Vec2f() : geometry::Vec2f(10.f, 10.f);
+    }
+
+    /// Returns the icon size overrides.
+    ///
+    const geometry::Vec2f& iconSizeOverrides() const {
+        return iconSizeOverrides_;
+    }
+
+    /// Sets the icon size overrides.
+    /// A component value < 0.f means it is automatic (uses preferred-size).
+    /// Any component value at 0.f makes the icon invisible.
+    ///
+    /// No geometry update request is made.
+    ///
+    void setIconSizeOverrides(float x, float y) {
+        iconSizeOverrides_ = geometry::Vec2f(x, y);
+    }
+
+    /// Returns the shortcut size overrides.
+    ///
+    const geometry::Vec2f& shortcutSizeOverrides() const {
+        return shortcutSizeOverrides_;
+    }
+
+    /// Sets the shortcut size overrides.
+    /// A component value < 0.f means it is automatic (uses preferred-size).
+    /// Any component value at 0.f makes the shortcut invisible.
+    ///
+    /// No geometry update request is made.
+    ///
+    void setShortcutSizeOverrides(float x, float y) {
+        shortcutSizeOverrides_ = geometry::Vec2f(x, y);
+    }
+
+    Menu* parentMenu() const {
+        return parentMenu_;
+    };
+
+    Menu* menuPopup() const {
```

Comment:
Shouldn't this be `popupMenu`, since it returns a menu? I agree on the other though: `closeMenuPopup` makes sense (it closes the popup of the menu), as well as the `menuPopupOpened/Closed`.